app/models/hardware.rb

```diff
@@ -63,8 +63,14 @@ def self.add_elements(parent, xmlNode)
                    .select(:id, :device_type, :location, :address)
                    .collect { |rec| [rec.id, [rec.device_type, rec.location, rec.address]] }
 
-    deletes[:disk] = parent.hardware.disks.select(:id, :device_type, :location)
+    if parent.vendor == "redhat"
+      deletes[:disk] = parent.hardware.disks.select(:id, :device_type, :location)
+                     .collect { |rec| [rec.id, [rec.device_type, "0:#{rec.location}"]] }
```

Comment:
As we discussed, if we make this change in RHEV specific vmConfig hash, it will break SSA at other place. Any changes after vmConfig is populated have to have this "redhat" specific check. Maybe we should find a better way to compare disks, rather than depending on its location. But that needs more time and works.